presto-native-execution/presto_cpp/main/PrestoTask.cpp

```diff
@@ -395,49 +460,17 @@ protocol::TaskInfo PrestoTask::updateInfoLocked() {
         }
       }
       if (op.numSplits != 0) {
-        const auto statName = fmt::format(
-            "{}.{}.{}", op.operatorType, op.planNodeId, "numSplits");
+        const auto statName =
+            fmt::format("{}.{}.numSplits", op.operatorType, op.planNodeId);
         opOut.runtimeStats.emplace(
-            statName,
-            protocol::RuntimeMetric{
-                statName,
-                protocol::RuntimeUnit::NONE,
-                op.numSplits,
-                1,
-                op.numSplits,
-                op.numSplits});
+            statName, createProtocolRuntimeMetric(statName, op.numSplits));
       }
 
-      // If Velox's operator has spilling stats, then add them to the protocol
+      addRunningTimeOperatorMetrics(opOut, prestoTaskStats, op);
+      // If Velox operator has spilling stats, then add them to the Presto
       // operator stats and the task stats as runtime stats.
       if (op.spilledBytes > 0) {
-        std::string statName =
-            fmt::format("{}.{}.spilledBytes", op.operatorType, op.planNodeId);
-        auto protocolMetric = createProtocolRuntimeMetric(
-            statName, op.spilledBytes, protocol::RuntimeUnit::BYTE);
-        opOut.runtimeStats.emplace(statName, protocolMetric);
-        prestoTaskStats.runtimeStats[statName] = protocolMetric;
-
-        statName =
-            fmt::format("{}.{}.spilledRows", op.operatorType, op.planNodeId);
-        protocolMetric = createProtocolRuntimeMetric(
-            statName, op.spilledRows, protocol::RuntimeUnit::NONE);
-        opOut.runtimeStats.emplace(statName, protocolMetric);
-        prestoTaskStats.runtimeStats[statName] = protocolMetric;
-
-        statName = fmt::format(
-            "{}.{}.spilledPartitions", op.operatorType, op.planNodeId);
-        protocolMetric = createProtocolRuntimeMetric(
-            statName, op.spilledPartitions, protocol::RuntimeUnit::NONE);
-        opOut.runtimeStats.emplace(statName, protocolMetric);
-        prestoTaskStats.runtimeStats[statName] = protocolMetric;
-
-        statName =
-            fmt::format("{}.{}.spilledFiles", op.operatorType, op.planNodeId);
-        protocolMetric = createProtocolRuntimeMetric(
-            statName, op.spilledFiles, protocol::RuntimeUnit::NONE);
-        opOut.runtimeStats.emplace(statName, protocolMetric);
-        prestoTaskStats.runtimeStats[statName] = protocolMetric;
+        addSpillingOperatorMetrics(opOut, prestoTaskStats, op);
```

Comment:
It looks we use the pattern that for task-level runtime stats inside the loop we update the variable `taskRuntimeStats`, ex.
https://github.com/prestodb/presto/blob/089c06f68644aa63efad0bac8dc150a62c2460f3/presto-native-execution/presto_cpp/main/PrestoTask.cpp#L408-L411

Then outside the loop we copy `taskRuntimeStats` to `prestoTaskStats.runtimeStats` altogether
https://github.com/prestodb/presto/blob/089c06f68644aa63efad0bac8dc150a62c2460f3/presto-native-execution/presto_cpp/main/PrestoTask.cpp#L497-L500

Feeling it's better to follow this pattern to be consistent. So we can pass `taskRuntimeStats` here. There're quite a few runtime variables in this func, difficult for me to tell how they're different :).

Reply:
This function merely replaces the code that was there.
I believe it was designed that way so we don't need to create Velox runtime stats and the protocol runtime stats.
```taskRuntimeStats``` holds velox structure, while when we fill up opOut.runtimeStats we create protocol structure.
So in order not two create both velox and protocol structures we take just created protocol structure and add it directly to the ```prestoTaskStats``` instead of using ```taskRuntimeStats```, which would be converted to ```prestoTaskStats``` afterwards.

Reply:
I got it.